bin/jruby.sh

```diff
@@ -628,6 +633,11 @@ fi
 # The rest of the arguments are for ruby
 append ruby_args "$@"
 
+# If regenerating the JSA archive but no Ruby arguments were passed, do -e 1
+if $regenerate_jsa_file & [ \'"$ruby_args"\' = "' '" ]; then
+    assign ruby_args '-e' '1'
+fi
```

Comment:
That looks like a pretty fragile way to check that the `ruby_args` array is empty, I would prefer to either parse it and check `$#` or we could get clever and match against `*[![:space:]]*`. Also quotes are unnecessary on line 638

Reply:
Oh and there's a bug here, `&` forks off a subshell so if `[ "$ruby_args" = " " ]` then this will always execute